lib/astarte_core/triggers/policy/policy.ex

```diff
@@ -0,0 +1,185 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Triggers.Policy do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Astarte.Core.Triggers.Policy
+  alias Astarte.Core.Triggers.Policy.Handler
+  alias Astarte.Core.Triggers.PolicyProtobuf.Policy, as: PolicyProto
+
+  @policy_name_regex ~r/^(?!@).+$/
+
+  @required_fields [
+    :name,
+    :error_handlers,
+    :maximum_capacity
+  ]
+
+  @permitted_fields [
+    :name,
+    :maximum_capacity,
+    :retry_times,
+    :event_ttl
+  ]
+
+  @derive Jason.Encoder
+  @primary_key false
+  embedded_schema do
+    field :name
+    field :maximum_capacity, :integer
+    field :retry_times, :integer
+    field :event_ttl, :integer
+    embeds_many :error_handlers, Policy.Handler
+  end
+
+  def changeset(%Policy{} = policy, params \\ %{}) do
+    # TODO check minimum and maximum range of attributes + conditions on name
+    policy
+    |> cast(params, @permitted_fields)
+    |> validate_required(@required_fields)
+    |> validate_length(:name, max: 128)
+    |> validate_format(:name, @policy_name_regex)
+    |> validate_inclusion(:retry_times, 1..100)
+    |> validate_inclusion(:event_ttl, 1..86_400)
+    |> validate_inclusion(:maximum_capacity, 1..1_000_000)
+    |> cast_embed(:error_handlers, with: &Handler.changeset/2, required: true)
+    |> validate_length(:error_handlers, min: 1, max: 200)
+    |> validate_all_handlers_on_different_errors()
+    |> validate_retry_times_compatible()
+  end
+
+  def valid_name?(name) do
+    String.match?(name, @policy_name_regex)
+  end
+
+  defp validate_all_handlers_on_different_errors(changeset) do
+    handlers = get_field(changeset, :error_handlers, [])
+
+    if disjoint_errors?(handlers) do
+      changeset
+    else
+      add_error(changeset, :error_handlers, "must all handle distinct errors.")
+    end
+  end
+
+  # TODO Maybe use a computationally better comparison?
```

Comment:
This TODO can be removed since I don't think this can be optimized further, at least for now